### DISCUSSION TITLE
DOI metadata in REST API requests should be easier to externally and locally validate

### DISCUSSION THREAD

**Discussion by codycooperross** [datacite/datacite-suggestions#80] [General Suggestions]
### What is the problem that your suggestion solves?

There is no way to  externally and locally validate the JSON DOI metadata in a REST API request before sending it. 

### What solution might meet your needs?

Resources should be available to  externally and locally validate the metadata in a REST API request. 

### What alternatives have you tried or considered?

Using the documentation on [constructing a JSON payload](https://support.datacite.org/docs/api-create-dois#constructing-the-json-payload) and the [DataCite REST API Reference](https://support.datacite.org/reference/post_dois). Sending requests to the REST API to determine if they receive a 422 response for invalid metadata. 

### Is there anything else you would like to share?

[Link for DataCite staff](https://app.frontapp.com/open/msg_itm3sut?key=Gy9cErkOLUojn-9BaFsdFjQ-vhWLe6MB)
[Link for DataCite staff
](https://app.frontapp.com/open/cnv_b8n85p1?key=t2S7N9G_Do7sXbkdI_6LVPnjP0xNbveU)[Link for DataCite staff
](https://docs.google.com/spreadsheets/d/16jQnuW_DMgbb-74Y5TByUG4IkHdvJSZ0Fe_qxQ4Xl6w/edit?gid=444055379#gid=444055379&range=17:17)
### What group(s) would benefit from your suggestion?

- [X] DataCite members registering DOIs and metadata and administering accounts
- [ ] Developers building integrations with DataCite services
- [ ] Harvesters who index DataCite DOIs and metadata
- [ ] Individuals or institutions analyzing DataCite DOIs and metadata
- [ ] DataCite staff
- [ ] Other

**Comment by yarikoptic** [comment]
FWIW: In our case [we validate](https://github.com/dandi/dandi-schema/blob/782c421e60b5989d3586d9cf89c526b1ac0a6778/dandischema/datacite/__init__.py#L258) our records to be submitted against JSON Schemas [provided  by inveniosoftware](https://github.com/inveniosoftware/datacite/tree/master/datacite/schemas). Related discussions/issues:
- https://github.com/datacite/schema/issues/149
- https://github.com/inveniosoftware/datacite/issues/101
- https://github.com/inveniosoftware/datacite/issues/102#issuecomment-2486657969
